subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c

```diff
@@ -45,103 +51,167 @@ os_mgmt_echo(struct mgmt_ctxt *ctxt)
 		return MGMT_ERR_EINVAL;
 	}
 
-	err |= cbor_encode_text_stringz(&ctxt->encoder, "r");
-	err |= cbor_encode_text_string(&ctxt->encoder, echo_buf, strlen(echo_buf));
+	err = cbor_encode_text_stringz(&ctxt->encoder, "r")				||
+	      cbor_encode_text_string(&ctxt->encoder, echo_buf, strlen(echo_buf));
 
-	if (err != 0) {
-		return MGMT_ERR_ENOMEM;
-	}
 
-	return 0;
+	return (err == 0) ? 0 : MGMT_ERR_ENOMEM;
 }
 #endif
 
 #if OS_MGMT_TASKSTAT
+static inline int
+os_mgmt_taskstat_encode_thread_name(struct CborEncoder *encoder, int idx,
+				    const struct k_thread *thread)
+{
+	int err = 0;
+
+#ifdef CONFIG_THREAD_NAME
+	ARG_UNUSED(idx);
+	err = cbor_encode_text_string(encoder, thread->name,
+			strnlen(thread->name, CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN - 1));
+#else
+	char thread_name[CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN];
+
+	thread_name[sizeof(thread_name) - 1] = 0;
+#ifndef CONFIG_OS_MGMT_TASKSTAT_USE_THREAD_IDX_FOR_NAME
+	ARG_UNUSED(idx);
+	ll_to_s((int)thread->base.prio, sizeof(thread_name) - 1, thread_name);
+#else
+	ARG_UNUSED(thread);
+	ll_to_s(idx, sizeof(thread_name) - 1, thread_name);
+#endif /* CONFIG_OS_MGMT_TASKSTAT_USE_THREAD_IDX_FOR_NAME */
+	err = cbor_encode_text_stringz(encoder, thread_name);
+#endif /* CONFIG_THREAD_NAME */
+
+	return err;
+}
+
+static inline int
+os_mgmt_taskstat_encode_stack_info(struct CborEncoder *thread_map,
+				   const struct k_thread *thread)
+{
+#ifdef CONFIG_OS_MGMT_TASKSTAT_STACK_INFO
+	ssize_t stack_size = 0;
+	ssize_t stack_used = 0;
+	int err = 0;
+
+#ifdef CONFIG_THREAD_STACK_INFO
+	stack_size = thread->stack_info.size / 4;
+
+#ifdef CONFIG_INIT_STACKS
+	unsigned int stack_unused;
+
+	if (k_thread_stack_space_get(thread, &stack_unused) == 0) {
+		stack_used = thread->stack_info.size - stack_unused;
```

Comment:
Using /4 for size but no divisor for used

Reply:
Good catch.

Reply:
Done